src/sync/sync-api.js

```diff
@@ -50,11 +71,60 @@ export class SyncApi extends TypedEmitter {
   #pscByPeerId = new Map()
   /** @type {Set<string>} */
   #peerIds = new Set()
-  #isSyncing = false
+  #autostopper = new Autostopper(this.#autostop.bind(this))
+  /** @type {InternalSyncState} */
+  #internalSyncState = {
+    enabledNamespaceGroups: 'initial only',
+    autostopAfter: Infinity,
+  }
   /** @type {Map<import('protomux'), Set<Buffer>>} */
   #pendingDiscoveryKeys = new Map()
   #l
 
+  /** @returns {'none' | 'initial only' | 'all'} */
+  get #enabledNamespaceGroups() {
+    return 'enabledNamespaceGroups' in this.#internalSyncState
+      ? this.#internalSyncState.enabledNamespaceGroups
+      : 'none'
+  }
+
+  /**
+   * @returns {boolean}
+   */
+  get #isSyncingData() {
+    const enabledNamespaceGroups = this.#enabledNamespaceGroups
+    switch (enabledNamespaceGroups) {
+      case 'none':
+      case 'initial only':
+        return false
+      case 'all':
+        return true
+      default:
+        throw new ExhaustivenessError(enabledNamespaceGroups)
+    }
+  }
+
+  #update() {
```

Comment:
Previously, `start()` and `stop()` modified the `PeerSyncController`s directly. Now, you describe what you want in `#internalSyncState` and apply it with `#update()`. You should be able to call `#update()` whenever you want.

Note that _logging was removed_ because it was hard to apply that here. Let me know if I should re-add that back.